1st chunk: allow to install js plugin without having gjs at configure time
cf https://gitlab.gnome.org/GNOME/gimp/-/issues/12424

Index: meson.build
--- meson.build.orig
+++ meson.build
@@ -1124,7 +1124,7 @@ MIMEtypes += 'image/openraster'
 
 ## Javascript
 
-gjs = find_program('gjs', required: get_option('javascript'))
+gjs = find_program('gjs', required: false)
 have_javascript = get_option('javascript').enabled() or (gjs.found() and get_option('javascript').auto())
 if not gjs.found() and have_javascript
   gjs = find_program('gjs', required: false)
@@ -1149,7 +1149,7 @@ have_lua_output = have_lua
 if have_lua
   have_lua_lgi = false
 
-  foreach lua_bin : [ 'luajit', 'lua5.1', 'lua-5.1', 'lua' ]
+  foreach lua_bin : [ 'luajit${LUAJIT_V}', 'lua5.1', 'lua-5.1', 'lua' ]
     lua = find_program(lua_bin, required: false)
 
     if lua.found() and meson.can_run_host_binaries()
@@ -1533,6 +1533,7 @@ conf.set('NO_FD_SET',
   not platform_windows
   and not cc.compiles('''
     #include <sys/types.h>
+    #include <sys/select.h>
     int main() { fd_set readMask, writeMask; return 0; }
   ''')
 )
